static/app/utils/string/ellipsize.tsx

```diff
@@ -0,0 +1,24 @@
+/**
+ * Truncates a string to a maximum length and adds an ellipsis character (…) if truncated. Similar to the Python `truncatechars`. Does not trim the string, please trim before calling this function. Does right-trim the string before output to avoid having whitespace followed by an ellipsis.
+ *
+ * @example
+ * ellipsize('hello world', 8) // returns 'hello wo…'
+ * ellipsize('hello world', 6) // returns 'hello…'
+ * ellipsize('    ', 6) // returns ''
+ * ellipsize('short', 10) // returns 'short'
+ */
+export function ellipsize(str: string, maxLength: number): string {
+  if (maxLength < 0) {
+    return '';
+  }
+
+  if (str.trim().length === 0) {
+    return '';
+  }
+
+  if (str.length > maxLength) {
+    return str.slice(0, maxLength).trimRight() + '\u2026';
+  }
```

Comment:
Proposed flow change to remove (the str.trim.length === 0) is and added a NaN check because calling `"test".slice(0, NaN)` seems to return `""`. 

The [spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice#description) doesn't mention indexEnd, but it does specify that if indexStart cannot be converted to a number, it is treated as 0. A console test seems to confirm that that is the case for indexEnd as well

```suggestion
export function ellipsize(str: string, length: number): string {
  if(length < 0 || isNaN(length)) throw new TypeError('Invalid string length argument value of ${length} provided to ellipsize');
   
  if (str.length < length) return str;
  return str.slice(0, maxLength).trimRight() + '\u2026';
```

Reply:
👀 great find, thanks! `NaN` is runtime poison, I wish TypeScript helped in this regard 😮‍💨